### PR TITLE
Add WITH_XML, WITH_YAML and WITH_JSON to mk_mingw.mak

### PR DESCRIPTION
**Pull request by mattn**
mk_mingw.mak does not have a way to build with +yaml or +xml

### CODE REVIEW

**Comment by blueyed** [comment]
# [Codecov](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=h1) Report
> Merging [#2440](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=desc) into [master](https://codecov.io/gh/universal-ctags/ctags/commit/39e72505745bd6ff44913361e19bd06782debb09&el=desc) will **not change** coverage by `%`.
> The diff coverage is `n/a`.

[Impacted file tree graph](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=tree)

```diff
@@           Coverage Diff           @@
##           master    #2440   +/-   ##
=======================================
  Coverage   86.57%   86.57%           
=======================================
  Files         177      177           
  Lines       36304    36304           
=======================================
  Hits        31431    31431           
  Misses       4873     4873           
```



------

[Continue to review full report at Codecov](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=continue).
> **Legend** - [Click here to learn more](https://docs.codecov.io/docs/codecov-delta)
> `Δ = absolute <relative> (impact)`, `ø = not affected`, `? = missing data`
> Powered by [Codecov](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=footer). Last update [39e7250...2b8af64](https://codecov.io/gh/universal-ctags/ctags/pull/2440?src=pr&el=lastupdated). Read the [comment docs](https://docs.codecov.io/docs/pull-request-comments).

**Comment by k-takata** [comment]
How about +json?

Don't you need to add `LIBS += -lxml2`, etc.?

**Comment by mattn** [comment]
Right, I seems mistake something. Will update.

File: mk_mingw.mak [hunk 1]
```diff
@@ -9,7 +9,7 @@ CFLAGS = -Wall -std=gnu99
 SIZE_T_FMT_CHAR='""'
 COMMON_DEFINES=-DUSE_SYSTEM_STRNLEN
 DEFINES = -DWIN32 $(REGEX_DEFINES) -DHAVE_PACKCC $(COMMON_DEFINES)
-INCLUDES = -I. -Imain -Ignu_regex -Ifnmatch -Iparsers
+INCLUDES = -I. -Ignu_regex -Ifnmatch -iquote parser -iquote main
```
Comment (k-takata) [inline mk_mingw.mak]: Maybe a mistake?
```suggestion
INCLUDES = -I. -Ignu_regex -Ifnmatch -iquote parsers -iquote main
```

**Comment by k-takata** [comment]
I haven't tested this by myself, but LGTM.

**Comment by k-takata** [comment]
(Off topic)
I have installed codecov app in this repo, and now @codecov-io reports the result.
@blueyed Maybe you can disable your codecov setting?

File: mk_mingw.mak (hunk 1 above)
Comment (masatake) [inline mk_mingw.mak]: It this related to using pkg-config?
If not but the change improves something, I would like you to make a separate commit.

**Comment by masatake** [comment]
If possible, I would like to keep the history clean; each commit should be complete by itself.

2b8af64 is incomplete because it requires the rest of commits.
How about reorganize your patches into 3 commits
* commit A. enabling libxml2,
* commit B. enabling libyaml, and
* commit C. enabling jasson.

So ctags with commit A works well even B and/or C are/is not applied.
So ctags with commit B works well even C and/or A are/is not applied.
So ctags with commit C works well even A and/or B are/is not applied.

Such complete patches is git blame friendly.

I'm not sure the change about `-iquote parser -iquote main` should be a separated commit or not.

Of course, during development, the commits can be incomplete.
However, I would like to say such commits are not ready for reviewing and merging.
This should be happens only in your PC for developing, or please, set "DRAFT" flag
of this pull request like https://github.com/universal-ctags/ctags/pull/2241. 
So you preevent me from reviewing your change seriously. 

git push --force is welcome.

**Comment by masatake** [comment]
Allow me adding more comments.

Such complete patches helps me in the future.
When we introduce a new library to u-ctags in the future, I study what I have to do from the compelete patch. If a conceptually one change spreads over multiple patches, understanding
what we should do becomes harder.

**Comment by masatake** [comment]
Thank you very much.
